resources/aws/security_group.go

```diff
@@ -1,49 +1,38 @@
 package aws
 
 import (
-	"strconv"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
 	microerror "github.com/giantswarm/microkit/error"
 )
 
 type SecurityGroup struct {
 	Description string
 	GroupName   string
 	VpcID       string
-	PortsToOpen []int
+	Rules       Rules
 	id          string
 	AWSEntity
 }
 
-func (s SecurityGroup) findExisting() (*ec2.SecurityGroup, error) {
-	portRuleFilters := []*ec2.Filter{
-		&ec2.Filter{
-			Name: aws.String(subnetVpcID),
-			Values: []*string{
-				aws.String(s.VpcID),
-			},
-		},
-	}
+// Rule is a Security Group rule.
+type Rule struct {
+	Port int
+	// SourceCIDR is the CIDR of the source. Conflicts with SourceCIDR.
+	SourceCIDR string
+	// SecurityGroupID is the ID of the Security Group. Conflicts with SecurityGroupID.
+	SecurityGroupID string
+}
 
-	for _, port := range s.PortsToOpen {
-		portRuleFilters = append(portRuleFilters, &ec2.Filter{
-			Name: aws.String(securityGroupIPPermissionFromPort),
-			Values: []*string{
-				aws.String(strconv.Itoa(port)),
-			},
-		}, &ec2.Filter{
-			Name: aws.String(securityGroupIPPermissionToPort),
-			Values: []*string{
-				aws.String(strconv.Itoa(port)),
-			},
-		})
-	}
+// Rules is a slice of Rule structs.
+type Rules []Rule
```

Comment:
IMO it is only useful to define a type for a list in case there is some functionality bound to the list type. I do not see that there is any method being implemented on `Rules`. I would then like to simply define `[]Rule` where it is used.